build_tools/bazel_to_cmake/bazel_to_cmake_converter.py

```diff
@@ -893,9 +878,27 @@ def GetDict(obj):
   return ret
 
 
-def convert_build_file(build_file_code, allow_partial_conversion=False):
+def convert_build_file(build_file_code,
+                       repo_cfg,
+                       allow_partial_conversion=False):
   converter = Converter()
-  exec(build_file_code, GetDict(BuildFileFunctions(converter)))
+  # Allow overrides of TargetConverter and BuildFileFunctions from repo cfg.
+  if hasattr(repo_cfg, "REPO_MAP"):
+    repo_map = repo_cfg.REPO_MAP
+  else:
+    repo_map = {}
+  if hasattr(repo_cfg, "CustomTargetConverter"):
+    target_converter = repo_cfg.CustomTargetConverter(repo_map=repo_map)
+  else:
+    target_converter = bazel_to_cmake_targets.TargetConverter()
+  if hasattr(repo_cfg, "CustomBuildFileFunctions"):
+    build_file_functions = repo_cfg.CustomBuildFileFunctions(
+        converter=converter, targets=target_converter)
+  else:
+    build_file_functions = BuildFileFunctions(converter=converter,
+                                              targets=target_converter)
+
```

Comment:
Use `getattr` with default instead? Also, shouldn't we pass `repo_map` to the default `TargetConverter` as well?

```suggestion
  repo_map = getattr(repo_cfg, "REPO_MAP", {})
  target_converter = getattr(repo_cfg, "CustomTargetConverter", bazel_to_cmake_targets.TargetConverter)(repo_map)
  build_file_functions = getattr(repo_cfg, "CustomBuildFileFunctions", BuildFileFunctions)(converter=converter, targets=target_converter)
```

Reply:
Thanks.